Project/Tsofen/Job.py

```diff
@@ -0,0 +1,97 @@
+import json
+import os
+import re
+import enums
+from enums import Status
+
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+import logging
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
+class Job:
+    def __init__(self):
+        """
+        constructor
+        """
+        self._email_addresses=None
+        self.__valid_emails=[]
+        self.__invalid_emails = []
+        self._email_file_name=None
+        self._data = None
+        self.__email_sender='awawdy.ahmad@gmail.com'
+        self.__password = 'sbgg wwwp ovft chsc'
+    def _prerequisite(self):
+        """
+        check emails validation
+        """
+        try:
+            email_address_format = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
+            self._data=self._load_json_file(self._email_file_name)
+            list_domains=self._data["domains"]
+            email_addresses_list=str(self._email_addresses).split(',')
+            for addr in email_addresses_list:
+                if re.match(email_address_format,addr):
+                    if addr[addr.find('@'):] in list_domains:
+                        self.__valid_emails.append(addr)
+                    else:
+                        self.__invalid_emails.append(addr)
+                else:
+                    logging.warning(f'Job._prerequisite: {addr} is invalid email format')
+        except Exception as e:
+            raise Exception(e)
+
+    def _send_summarization_email(self,status,send_email):
+        """
+        send emails to valid addresses
+        """
+        try:
+            if send_email:
+                if status is Status.Success.value:
+                    body = "The comparison has succeeded"
+                else:
+                    body = "The comparison has failed"
+                if len(self.__valid_emails) == 0 and len(self.__invalid_emails) == 0:
```

Comment:
```suggestion
                if len(self.__valid_emails) == 0 and len(self.__invalid_emails) == 0:
```
please try to move these checks into prerequisite